src/array/cuda/atomic.cuh

```diff
@@ -304,16 +304,16 @@ __device__ __forceinline__ half AtomicAdd<half>(half* addr, half val) {
   printf(
       "Atomic operations are not supported for half precision (FP16) "
       "on this GPU.\n");
-  __trap();
+  abort();
   return val;
 #endif  // __CUDA_ARCH__ >= 700
 }
 #endif  // defined(CUDART_VERSION) && CUDART_VERSION >= 10000
 
 #if BF16_ENABLED
 template <>
-__device__ __forceinline__ __nv_bfloat16
-AtomicAdd<__nv_bfloat16>(__nv_bfloat16* addr, __nv_bfloat16 val) {
+__device__ __forceinline__ __hip_bfloat16
+AtomicAdd<__hip_bfloat16>(__hip_bfloat16* addr, __hip_bfloat16 val) {
 // make sure we have bfloat16 support
 #if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 800
```

Comment:
hipify can't automatically handle things like `__CUDA_ARCH__`.  Is this PR trying to address these cases manually?